Adding "Cloud Native Development" pre-engagement guide. (#997)
vmware-tanzu/tanzu-dev-portal: content/outcomes/cloud-native-development/specific__user_interfaces.md

---
title: User Interfaces
weight: 180
layout: single
team:
 - VMware Tanzu Labs
---

Traditional Java Enterprise Edition (EE) web applications evolved through many model-view-controller technology stacks, including templating engines such as Java Server Pages (JSP), Java Server Faces (JSF), Apache Struts, or Portlets. Each of these stacks tried to tackle the separation between server-based business logic and the views themselves while attempting to break up the UI into modular components. The modular components are tightly coupled to an underlying heavyweight server environment. Applications that use these view technologies are often prime candidates for more extensive modernization efforts, as they tend to crystalize in monolithic web applications with many functions and layers.

> Depending on the current state of the application and the amount of refactoring your team wants to take on, we may port the existing JSP or similar view templates over to deploy your application in a TAS Tomcat container. This provides a simple and non-intrusive way to get one of these technologies running on the platform.

Reviewer: ```suggestion
Depending on the current state of the application and the amount of refactoring the team wants to take on, we suggest porting the existing JSP or similar view templates over to deploy an application in a TAS Tomcat container. This provides a simple and non-intrusive way to get one of these technologies running on the platform.
```

Reviewer: Making more generic
